### PR TITLE
Auth service

### PR DESCRIPTION
**Pull request by doraby**
Auth service:  
— Create Google auth service
— store user to the app component
— pass the user to the Nav and Main component
— add logout component
— Delete register functionality because it is not needed anymore
— create a new user after login
— create admin and beginner roles (0 admin, 4 beginners)
— protected route for admin and not authenticated  user
— Hide AnswerActions for unauthorized user??
— add loading
— Redirect to login page
— correct some errors in routes
— pass asnwerID to user
— pass user UId to answer

Refactor:
Delete some empty folders and replace some common components

to discuss:
— Add firebaseUI to the package but don't use it. How to delete it?

### CODE REVIEW

File: src/Components/ErrorPages/pageForbidden.jsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+const PageForbidden = () => {
+    return (
+        <div>
+            ololol
```
Comment (AlexDovgalenok) [inline src/Components/ErrorPages/pageForbidden.jsx]: Is it the final content?

File: src/Components/Profile/profile.jsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React, {Component} from 'react';
+
+class Profile extends Component {
+    render() {
+        return (
+            <div>
+                fjdksljfa;jfkl;da
```
Comment (AlexDovgalenok) [inline src/Components/Profile/profile.jsx]: ?

File: src/Components/ErrorPages/pageForbidden.jsx (hunk 1 above)
Comment (doraby) [reply src/Components/ErrorPages/pageForbidden.jsx]: no it's just placeholder %)

File: src/Components/Profile/profile.jsx (hunk 2 above)
Comment (doraby) [reply src/Components/Profile/profile.jsx]: It's a placeholder

**Pull request merged**